Modix/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -33,6 +34,7 @@ public static IServiceCollection AddModix(this IServiceCollection services)
                 }));
 
             services.AddSingleton<IDiscordClient>(provider => provider.GetRequiredService<DiscordSocketClient>());
+            services.AddScoped<ISelfUser>(p => p.GetRequiredService<DiscordSocketClient>().CurrentUser);
```

Comment:
Shouldn't we be able to make this a `Singleton`? There's only one bot user, and it's not going to change in the lifetime of the program. Or is there a thread-safety issue?

Reply:
It's more of a race condition, as this won't resolve properly until DiscordSocketClient.Ready has fired, I believe

Reply:
That's fair. I'm still not a fan of this being a registered service at all, when you could just request `IDiscordClient` and get it yourself, but it is a bit handy, I guess.

Reply:
I want to leave the ISelfUser binding as its own thing. It feels more decoupled to me that way and I would imagine it'd make testing easier rather than having to mock a DiscordClient when all the handler cares about is the user

Reply:
I wasn't planning on arguing the point.